DQM/HLTEvF/plugins/LumiMonitor.cc

```diff
@@ -0,0 +1,246 @@
+  , minPixelClusterCharge_        ( iConfig.getParameter<double>     ("minPixelClusterCharge")       )
+{
+
+  numberOfPixelClustersVsLS_   = nullptr;
+  numberOfPixelClustersVsLumi_ = nullptr;
+  lumiVsLS_                    = nullptr;
+  pixelLumiVsLS_               = nullptr;
+  pixelLumiVsLumi_             = nullptr;
+
+  if(useBPixLayer1_) 
+    lumi_factor_per_bx_ = GetLumi::FREQ_ORBIT * GetLumi::SECONDS_PER_LS / GetLumi::XSEC_PIXEL_CLUSTER  ;
+  else
+    lumi_factor_per_bx_ = GetLumi::FREQ_ORBIT * GetLumi::SECONDS_PER_LS / GetLumi::rXSEC_PIXEL_CLUSTER  ;
+
+  edm::ParameterSet histoPSet        = iConfig.getParameter<edm::ParameterSet>("histoPSet");
+  edm::ParameterSet pixelClusterPSet = histoPSet.getParameter<edm::ParameterSet>("pixelClusterPSet");
+  edm::ParameterSet lumiPSet         = histoPSet.getParameter<edm::ParameterSet>("lumiPSet");
+  edm::ParameterSet pixellumiPSet    = histoPSet.getParameter<edm::ParameterSet>("pixellumiPSet");
+  edm::ParameterSet lsPSet           = histoPSet.getParameter<edm::ParameterSet>("lsPSet");
+
+  /*
+  pixelCluster_binning_(pixelClusterPSet.getParameter<int32_t>("nbins"), pixelClusterPSet.getParameter<double>("xmin"), pixelClusterPSet.getParameter<double>("xmax"));
+  lumi_binning_        (lumiPSet.getParameter<int32_t>        ("nbins"), lumiPSet.getParameter<double>        ("xmin"), lumiPSet.getParameter<double>        ("xmax"));
+  pixellumi_binning_   (pixellumiPSet.getParameter<int32_t>   ("nbins"), pixellumiPSet.getParameter<double>   ("xmin"), pixellumiPSet.getParameter<double>   ("xmax"));
+  ls_binning_          (lsPSet.getParameter<int32_t>          ("nbins"), lsPSet.getParameter<double>          ("xmin"), lsPSet.getParameter<double>          ("xmax"));
+  */
+
+  getHistoPSet  (pixellumiPSet,   pixellumi_binning_);
+  getHistoPSet  (pixelClusterPSet,pixelCluster_binning_);
+  getHistoPSet  (lumiPSet,        lumi_binning_);
+  getHistoLSPSet(lsPSet,          ls_binning_);
+
+  /*
+  pixellumi_binning_.nbins = pixellumiPSet.getParameter<int32_t>("nbins");
+  pixellumi_binning_.xmin  = pixellumiPSet.getParameter<double>("xmin");
+  pixellumi_binning_.xmax  = pixellumiPSet.getParameter<double>("xmax");
+
+  pixelCluster_binning_.nbins = pixelClusterPSet.getParameter<int32_t>("nbins");
+  pixelCluster_binning_.xmin  = pixelClusterPSet.getParameter<double>("xmin");
+  pixelCluster_binning_.xmax  = pixelClusterPSet.getParameter<double>("xmax");
+
+  lumi_binning_.nbins = lumiPSet.getParameter<int32_t>("nbins");
+  lumi_binning_.xmin  = lumiPSet.getParameter<double>("xmin");
+  lumi_binning_.xmax  = lumiPSet.getParameter<double>("xmax");
+
+  ls_binning_.nbins = lsPSet.getParameter<int32_t>("nbins");
+  ls_binning_.xmin  = lsPSet.getParameter<double>("xmin");
+  ls_binning_.xmax  = lsPSet.getParameter<double>("xmax");
+  */
+}
+
+void LumiMonitor::getHistoPSet(edm::ParameterSet& pset, MEbinning& mebinning)
```

Comment:
Can you change this to take (only) an `edm::ParameterSet const &` as input, and return a MEbinning by value ?
Something like:
```c++
MEbinning LumiMonitor::getHistoPSet(edm::ParameterSet const & pset)
{
  return MEbinning {
    pset.getParameter<int32_t>("nbins"),
    pset.getParameter<double>("xmin"),
    pset.getParameter<double>("xmax") };
}
```

This way, you can call it directly in the class initializer list.